src/Core/HmacClientHandler.cs

```diff
@@ -81,6 +83,23 @@ private static void SetHeader(HttpRequestMessage request, string name, string va
             request.Headers.Add(name, value);
         }
 
+        private static Uri GetRequestUri(HttpRequestMessage request)
+        {
+            if (string.IsNullOrWhiteSpace(request.Headers.Host))
+            {
+                return request.RequestUri;
+            }
+
+            var builder = new UriBuilder(request.RequestUri)
+            {
+                Host = request.Headers.Host,
+                Port = -1
```

Comment:
Why the port is set to `-1`?

Reply:
Without Post= -1 https://10.162.34.196:443/hello -> https://app-dev.if.lv:443/hello, validation fails on server
With Post= -1 https://10.162.34.196:443/hello -> https://app-dev.if.lv/hello

Reply:
If the port was explicitly set, there would be a problem. This code will remove the port, and it will not work either.

Reply:
Removed